package/MDAnalysis/coordinates/GRO.py

```diff
@@ -456,20 +454,22 @@ def write(self, obj):
                     ))
 
             # Footer: box dimensions
-            if np.allclose(ag.dimensions[3:], [90., 90., 90.]):
-                box = self.convert_pos_to_native(
-                    ag.dimensions[:3], inplace=False)
+            if (ag.dimensions is None or
+                np.allclose(ag.dimensions[3:], [90., 90., 90.])):
+                if ag.dimensions is None:
+                    box = np.zeros(3)
```

Comment:
Is this the behaviour we want? Tagging @jbarnoud here.

Reply:
I see, in any case, no matter what we do here we need to warn users of what we are doing.

Personally I'm still in the "if there's no standardised way to deal with this, then crash out" (folks can always add their own dimensions at their own discretion). But I'm happy with whatever the consensus is.

Reply:
See that at this point in the code we're just dealing with the writing of box coordinates when `ts.dimensions` is `None`. And as long as there's no tracking of input box validity, I don't think writing `ts.dimensions` that are `None` should ever be a problem to warn users about -- unless a given Writer has no canonical or sensible way to represent the no-box case (and the GROWriter does have).

For the record, while I don't oppose changing Readers to be more permissive with invalid boxes, and tracking input validity, I am also not for it. It may bring robustness, but do we really want to raise sloppy users that can't format their inputs right? Invalid boxes may even become mistakenly seen as a sanctioned extension to input formats. Stuff for other discussions.